modules/sdk/src/main/scala/org/tessellation/sdk/domain/healthcheck/consensus/HealthCheckConsensus.scala

```diff
@@ -204,17 +209,20 @@ abstract class HealthCheckConsensus[
 
              def participate = participateInRound(proposal.key, proposal.roundId)
 
-             inProgressRound
-               .map(_.processProposal(proposal))
-               .orElse {
-                 historicalRound.map(handleProposalForHistoricalRound(proposal))
-               }
-               .getOrElse {
-                 if (depth > 0)
-                   participate.flatMap(_ => handleProposal(proposal, depth - 1))
-                 else
-                   (new Throwable("Unexpected recursion!")).raiseError[F, Unit]
-               }
+             inProgressRound.map { r =>
+               r.hasProposal(proposal.owner)
```

Comment:
If it's exactly the same proposal then we can discard it immediately I guess (so the same observation and the same round id). If it's a proposal that matches the round id's but it's not exactly the same then it sounds like an error (node must have rejoined the consensus). If it's a different proposal yet then it's probably a proposal for the next round and then it should be put to a waiting proposals cache.

Reply:
Discarding is done within the round, in its `processProposal` function.

Reply:
I guess this one will be discarded the minute this round becomes historical. Makes sense as well, we don't need to remove it prematurely even if we know we could.

Reply:
Because if it's not passed to the round then it won't be discarded in round's `processProposal` function.